src/DotNetWorker.Core/FunctionMetadata/IRetryOptions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.Functions.Worker.Core.FunctionMetadata
+{
+    /// <summary>
+    /// Function execution retry policy to use on invocation failures.
+    /// </summary>
+    public interface IRetryOptions
+    {
+        /// <summary>
+        /// The maximum number of retries allowed per function execution. -1 means to retry indefinitely.
+        /// </summary>
+        public int? MaxRetryCount { get; }
+
+        /// <summary>
+        /// The delay used between retries when using a fixed delay strategy.
+        /// </summary>
+        public string? DelayInterval { get; }
+
+        /// <summary>
+        /// The minimum retry delay when using an exponential backoff strategy.
+        /// </summary>
+        public string? MinimumInterval { get; }
+
+        /// <summary>
+        /// The maximum retry delay when using an exponential backoff strategy.
+        /// </summary>
+        public string? MaximumInterval { get; }
+
+        /// <summary>
+        /// The retry strategy being used (fixed delay or exponential backoff).
+        /// </summary>
+        public string? Strategy { get; }
```

Comment:
Can we use an enum here?

Reply:
The grpc message uses an enum that's in the protofile, and I was thinking of waiting until the grpcworker layer to use that enum to parse the string here. The pattern being used to couple the grpc and core worker types is having them both implement an interface but for this particular property I was unsure how to go about it. 

I *think* we can still use the same pattern (have two enums, one in protofile and one in core implementing an interface defined in core) but wasn't 100% it was good practice.